### DISCUSSION TITLE
Error Cloning Repository After Generating New SSH Key in cpanel

### DISCUSSION THREAD

**Discussion by rhtshahi** [community/community#135831] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

In cpanel, I’m facing an error while trying to clone a private Git repository using a new SSH key. The first key I generated worked correctly, and I was able to clone the repository without issues. However, with the new key, which I am using for a different repository, I am encountering an error when attempting to clone. The repository URL is correct and exists, as it was successfully cloned with the previous key. 

Error: (XID eu8t8k) “/usr/local/cpanel/3rdparty/bin/git” reported error code “128” when it ended: ERROR: Repository not found. fatal: Could not read from remote repository. Please make sure you have the correct access rights and the repository exists. 

Is the problem related to github or single key limitation in cpanel? The discussion in this community post suggests that cPanel might only support one SSH key by default. 

For reference, here is the link to the discussion that addresses this potential limitation: https://support.cpanel.net/hc/en-us/community/posts/19154562063511-Manual-deployment-multiple-SSH-keys-issue-problem

**Comment by burningmannn** [comment]
By default, cPanel only supports one SSH key for Git authentication. When you generated the first key, it was likely added to the ~/.ssh/authorized_keys file, which allows cPanel to use that key for authentication. However, when you generated the new key for the different repository, it's not being used by cPanel because of this single-key limitation.

The error message "Repository not found" is misleading, as the repository does exist, and the URL is correct. The issue is that cPanel is not using the new SSH key to authenticate, so it's unable to access the repository.

To resolve this issue, you have a few options:

1. Use the same SSH key for both repositories: If you don't mind using the same key for both repositories, you can simply use the first key that worked. This way, you won't need to generate a new key or modify any settings in cPanel.
2. Add the new SSH key to the ~/.ssh/authorized_keys file: You can manually add the new SSH key to the ~/.ssh/authorized_keys file on your cPanel server. This will allow cPanel to use the new key for authentication. However, please note that this requires root access and some technical expertise.
3. Use a different authentication method: If you're using GitHub, you can use a personal access token instead of an SSH key. This way, you won't be limited by the single-key restriction in cPanel.
